verifier/sources/cpu/layout7/stark_verifier.move

```diff
@@ -0,0 +1,1073 @@
+
+    // constants
+    const PROOF_PARAMS_N_QUERIES_OFFSET: u64 = 0;
+    const PROOF_PARAMS_LOG_BLOWUP_FACTOR_OFFSET: u64 = 1;
+    const PROOF_PARAMS_PROOF_OF_WORK_BITS_OFFSET: u64 = 2;
+    const PROOF_PARAMS_FRI_LAST_LAYER_LOG_DEG_BOUND_OFFSET: u64 = 3;
+
+    struct ConstructorConfig has key, copy {
+        // The work required to generate an invalid proof is 2^numSecurityBits.
+        // Typical values: 80-128.
+        num_security_bits: u256,
+
+        // The secuirty of a proof is a composition of bits obtained by PoW and bits obtained by FRI
+        // queries. The verifier requires at least minProofOfWorkBits to be obtained by PoW.
+        // Typical values: 20-30.
+        min_proof_of_work_bits: u256
+    }
+
+    public fun init_stark_verifier(signer: &signer, num_security_bits: u256, min_proof_of_work_bits: u256) {
+        move_to(signer, ConstructorConfig {
+            num_security_bits,
+            min_proof_of_work_bits
+        });
+    }
+
+    public(friend) fun init_data_type(signer: &signer) {
+        move_to(signer, VpCheckpoint {
+            inner: CHECKPOINT1_VP
+        });
+        move_to(signer, CtxCache {
+            inner: vector[]
+        });
+        move_to(signer, CfflCheckpoint {
+            inner: CHECKPOINT1_CFFL
+        });
+        cpu_oods_7::init_data_type(signer);
+    }
+
+    // Adjusts the query indices and generates evaluation points for each query index.
+    // The operations above are independent but we can save gas by combining them as both
+    // operations require us to iterate the queries array.
+    //
+    // Indices adjustment:
+    //     The query indices adjustment is needed because both the Merkle verification and FRI
+    //     expect queries "full binary tree in array" indices.
+    //     The adjustment is simply adding evalDomainSize to each query.
+    //     Note that evalDomainSize == 2^(#FRI layers) == 2^(Merkle tree hight).
+    //
+    // evalPoints generation:
+    //     for each query index "idx" we compute the corresponding evaluation point:
+    //         g^(bitReverse(idx, log_evalDomainSize).
+    fun adjust_query_indices_and_prepare_eval_points(ctx: &mut vector<u256>) {
+        let n_unique_queries = (*borrow(ctx, MM_N_UNIQUE_QUERIES) as u64);
+        let fri_queue = MM_FRI_QUEUE;
+        let fri_queue_end = fri_queue + n_unique_queries * FRI_QUEUE_SLOT_SIZE;
+        let eval_points_ptr = MM_OODS_EVAL_POINTS;
+        let log_eval_domain_size = (*borrow(ctx, MM_LOG_EVAL_DOMAIN_SIZE) as u8);
+        let eval_domain_size = *borrow(ctx, MM_EVAL_DOMAIN_SIZE);
+        let eval_domain_generator = *borrow(ctx, MM_EVAL_DOMAIN_GENERATOR);
+
+        while (fri_queue < fri_queue_end) {
+            let query_idx = *borrow(ctx, fri_queue);
+            // Adjust queryIdx, see comment in function description.
+            let adjusted_query_idx = query_idx + eval_domain_size;
+            set_el(ctx, fri_queue, adjusted_query_idx);
```

Comment:
```suggestion
                  let query_idx = borrow_mut(ctx, fri_queue);
            // Adjust queryIdx, see comment in function description.
            *query_idx = *query_idx + eval_domain_size;

```

Reply:
The old `query_idx` will be used afterwards